### PR TITLE
Allows work pools for Prefect servers without API keys, and includes examples

### PR DESCRIPTION
**Pull request by chrisguidry**
We were inadvertently requiring an API key to connect to a remote Prefect
server, but that is only required for Prefect Cloud at this time.  This removes
that restriction, and adds examples of Prefect servers and Prefect Cloud
configurations for reference.

### CODE REVIEW

File: config/samples/v1_prefectworkpool_external_prefect_server.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+kind: PrefectWorkPool
+metadata:
+  labels:
+    app.kubernetes.io/name: prefect-operator
+    app.kubernetes.io/managed-by: kustomize
+  name: external-pool
+spec:
+  type: process
+  server:
+    remoteApiUrl: https://my-server.example.com
```
Comment (mitchnielsen) [inline config/samples/v1_prefectworkpool_external_prefect_server.yaml:11]: I noticed that we apply the `/api` suffix if one isn't provided, so I think it's fine to leave it out of the example - but I noticed that Prefect Cloud uses an `api` subdomain too: `api.prefect.cloud/api`. I'm guessing that's just a cloud-specific thing 🤔

**Review by mitchnielsen — APPROVED** [review]
Looks good 👍🏼 Picked an example config and got the external work pool set up against Prefect Cloud (dev):

image

File: config/samples/v1_prefectworkpool_external_prefect_server.yaml (hunk 1 above)
Comment (chrisguidry) [reply config/samples/v1_prefectworkpool_external_prefect_server.yaml:11]: Yep that's specific to Prefect Cloud, but folks who are hosting it themselves can use any domain they like (but it's always a subpath of `/api`)

File: api/v1/prefectworkpool_types.go [hunk 2]
```diff
@@ -143,11 +143,10 @@ func (s *PrefectWorkPool) Command() []string {
...
 }
 
-// PrefectAPIURL returns the API URL for the Prefect Server.
-// If an API Key is provided, it will return the RemoteAPIURL.
-// Otherwise, it will default to the local, in-cluster API URL.
+// PrefectAPIURL returns the API URL for the Prefect Server, either from the RemoteAPIURL or
+// from the in-cluster server
 func (s *PrefectWorkPool) PrefectAPIURL() string {
-	if s.Spec.Server.APIKey != nil && s.Spec.Server.RemoteAPIURL != nil {
+	if s.Spec.Server.RemoteAPIURL != nil {
```
Comment (parkedwards) [inline api/v1/prefectworkpool_types.go:149]: ah my fault, i was thinking of Cloud specifically for the remote access scenario

File: config/samples/v1_prefectworkpool_external_prefect_cloud.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+apiVersion: v1
```
Comment (parkedwards) [inline config/samples/v1_prefectworkpool_external_prefect_cloud.yaml:1]: dumb q, are these `samples/` used anywhere for the e2e tests? or are these just manifests for manual testing?

**Review by parkedwards — APPROVED** [review]

File: config/samples/v1_prefectworkpool_external_prefect_cloud.yaml (hunk 3 above)
Comment (chrisguidry) [reply config/samples/v1_prefectworkpool_external_prefect_cloud.yaml:1]: Oh great q I don't know about that, hmmm

Comment (mitchnielsen) [reply config/samples/v1_prefectworkpool_external_prefect_cloud.yaml:1]: Yeah don't think they're used anywhere, just there as examples for end users: https://book.kubebuilder.io/getting-started.html?highlight=samples#sample-of-custom-resources

Comment (mitchnielsen) [reply config/samples/v1_prefectworkpool_external_prefect_cloud.yaml:1]: They'd probably be helpful in CI when we want to do a full end-to-end test against a real instance, could `kubectl apply` these and confirm the resources come up sucessfully. Would also ensure our examples are functional

**Pull request merged**